src/stdio.rs

```diff
@@ -0,0 +1,243 @@
+use super::Error;
+
+use core::mem::ManuallyDrop;
+
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+use std::fs::File;
+use std::io::{self, IoSlice};
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+
+use std::process;
+
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, ReadBuf};
+
+#[derive(Debug)]
+pub(crate) enum StdioImpl {
+    /// Read/Write to /dev/null
+    Null,
+    /// Read/Write to a newly created pipe
+    Pipe,
+    /// Read/Write to custom fd
+    Fd(File),
+}
+
+/// Similar to std::process::Stdio
+#[derive(Debug)]
+pub struct Stdio(pub(crate) StdioImpl);
+impl Stdio {
+    /// Create a pipe for child communication
+    pub const fn piped() -> Self {
+        Self(StdioImpl::Pipe)
+    }
+
+    /// Pass /dev/null for child communication
+    pub const fn null() -> Self {
+        Self(StdioImpl::Null)
+    }
+}
+impl<T: IntoRawFd> From<T> for Stdio {
+    fn from(val: T) -> Self {
+        unsafe { Stdio::from_raw_fd(val.into_raw_fd()) }
+    }
+}
+impl FromRawFd for Stdio {
+    unsafe fn from_raw_fd(fd: RawFd) -> Self {
+        Self(StdioImpl::Fd(File::from_raw_fd(fd)))
+    }
+}
+impl From<Stdio> for process::Stdio {
+    fn from(stdio: Stdio) -> Self {
+        match stdio.0 {
+            StdioImpl::Null => process::Stdio::null(),
+            StdioImpl::Pipe => process::Stdio::piped(),
+            StdioImpl::Fd(fd) => unsafe {
+                process::Stdio::from_raw_fd(ManuallyDrop::new(fd).as_raw_fd())
```

Comment:
```suggestion
                process::Stdio::from_raw_fd(fd.into_raw_fd())
```
I _think_ this should be the same.

Reply:
Adopted.